c++/src/capnp/layout.h

```diff
@@ -656,6 +654,8 @@ class ListReader {
   Data::Reader asData();
   // Reinterpret the list as a blob.  Throws an exception if the elements are not byte-sized.
 
+  Data::Reader asDataOfAnySize();
```

Comment:
Hmm, perhaps this should be `kj::ArrayPtr<const byte> asRawBytes()`. That way `Data` specifically corresponds to the `Data` type used in Cap'n Proto schemas. This should also be documented as being valid only for data-only lists.

`getData()` in AnyList should similarly become `getRawBytes()` and return ArrayPtr.

`getDataSection()` in AnyStruct should also return ArrayPtr. (It should be safe to make this change.)